src/main/java/com/github/wvengen/maven/proguard/ProGuardMojo.java

```diff
@@ -183,216 +169,179 @@ public class ProGuardMojo extends AbstractMojo {
 	 *
 	 * <p>If {@link #attach} is 'true' the value is ignored and the name is constructed based on classifier.
-	 *
-	 * @parameter
 	 */
+	@Parameter
 	protected String outjar;
 
 	/**
 	 * Apply ProGuard classpathentry filters to output jar. e.g. <code>!**.gif,!**&#47;tests/**</code>
-	 *
-	 * @parameter
 	 */
+	@Parameter
 	protected String outFilter;
 
 	/**
 	 * Specifies whether to attach the created artifact to the project.
-	 *
-	 * @parameter default-value="false"
 	 */
+	@Parameter(defaultValue = "false")
 	private boolean attach;
 
 	/**
 	 * Determines if {@link #attach} also attaches the {@link #mappingFileName} file.
-	 *
-	 * @parameter default-value="false"
 	 */
+	@Parameter(defaultValue = "false")
 	private boolean attachMap;
 
 	/**
 	 * Determines if {@link #attach} also attaches the {@link #seedFileName} file.
-	 *
-	 * @parameter default-value="false"
 	 */
+	@Parameter(defaultValue = "false")
 	private boolean attachSeed;
 
 	/**
 	 * Specifies attach artifact type.
-	 *
-	 * @parameter default-value="jar"
 	 */
+	@Parameter(defaultValue = "jar")
 	private String attachArtifactType;
 
 	/**
 	 * Specifies attach artifact Classifier, ignored if {@link #attach} is 'false'.
-	 *
-	 * @parameter default-value="small"
 	 */
+	@Parameter(defaultValue = "small")
 	private String attachArtifactClassifier;
 
 	/**
 	 * Whether to append the {@link #attachArtifactClassifier} to the artifact final name.
-	 *
-	 * @parameter default-value="true"
 	 */
+	@Parameter(defaultValue = "true")
 	private boolean appendClassifier;
 
 	/**
 	 * Whether to include {@code META-INF/MANIFEST.MF} file
-	 *
-	 * @parameter default-value="false"
 	 */
+	@Parameter(defaultValue = "false")
 	private boolean addManifest;
 
 	/**
 	 * Whether to include {@code META-INF/maven/**} Maven descriptor.
-	 *
-	 * @parameter default-value="false"
 	 */
+	@Parameter(defaultValue = "false")
 	private boolean addMavenDescriptor;
 
 	/**
 	 * Directory containing the input and generated JAR.
-	 *
-	 * @parameter property="project.build.directory"
-	 * @required
 	 */
+	@Parameter(required = true, defaultValue = "${project.build.directory}")
 	protected File outputDirectory;
 
 	/**
 	 * The Maven project reference where the plugin is currently being executed. The default value is populated from
 	 * Maven.
-	 *
-	 * @parameter property="project"
-	 * @readonly
-	 * @required
 	 */
+	@Parameter(readonly = true, required = true, defaultValue = "${project}")
 	protected MavenProject mavenProject;
 
 	/**
 	 * The plugin dependencies.
-	 *
-	 * @parameter property="plugin.artifacts"
-	 * @required
-	 * @readonly
 	 */
+	@Parameter(readonly = true, required = true, defaultValue = "${plugin.artifacts}")
 	protected List<Artifact> pluginArtifacts;
 
-	/**
-	 * @component
-	 */
+	@Component
 	private MavenProjectHelper projectHelper;
 
 	/**
 	 * The Jar archiver.
-	 *
-	 * @component role="org.codehaus.plexus.archiver.Archiver" roleHint="jar"
-	 * @required
```

Comment:
Not sure if there is a way to set this 'required' now. Though maybe it had no effect before anyway?